eodag/rest/errors.py

```diff
@@ -0,0 +1,172 @@
+                )
+                or 500
+            )
+
+            if type(exception) in (MisconfiguredError, AuthenticationError):
+                logger.error("%s: %s", type(exception).__name__, str(exception))
+                error_dict[
+                    "message"
+                ] = "Internal server error: please contact the administrator"
+                error_dict.pop("detail", None)
+
+            if params := getattr(exception, "parameters", None):
+                for error_param in params:
+                    stac_param = EODAGSearch.to_stac(error_param)
+                    exception.message = exception.message.replace(
+                        error_param, stac_param
+                    )
+                error_dict["message"] = exception.message
+
+            error_list.append(error_dict)
+
+        return error_list
+
+    @property
+    def status_code(self) -> int:
+        """get global errors status code"""
+        if len(self._errors) == 1:
+            return self.errors[0]["status_code"]
+
+        return 400
+
+
+async def response_search_error_handler(
+    request: Request, exc: ResponseSearchError
+) -> ORJSONResponse:
+    """Handle ResponseSearchError exceptions"""
+    return ORJSONResponse(
+        status_code=exc.status_code,
+        content={"errors": exc.errors},
+    )
+
+
+async def eodag_errors_handler(request: Request, exc: EodagError) -> ORJSONResponse:
+    """Handler for EODAG errors"""
+    code = EODAG_DEFAULT_STATUS_CODES.get(type(exc), getattr(exc, "status_code", 500))
+    detail = f"{type(exc).__name__}: {str(exc)}"
+
+    if type(exc) in (MisconfiguredError, AuthenticationError, TimeOutError):
+        logger.error("%s: %s", type(exc).__name__, str(exc))
+
+    if type(exc) in (MisconfiguredError, AuthenticationError):
+        detail = "Internal server error: please contact the administrator"
+
+    if params := getattr(exc, "parameters", None):
+        for error_param in params:
+            stac_param = EODAGSearch.to_stac(error_param)
+            exc.message = exc.message.replace(error_param, stac_param)
+        detail = exc.message
+
+    return ORJSONResponse(
+        status_code=code,
+        content={"description": detail},
+    )
+
+
+def starlette_exception_handler(request: Request, error: Exception) -> ORJSONResponse:
+    """Default errors handle"""
+    description = (
+        getattr(error, "description", None)
+        or getattr(error, "detail", None)
+        or str(error)
+    )
+    return ORJSONResponse(
+        status_code=error.status_code,
+        content={"description": description},
+    )
+
+
+def add_exception_handlers(app: FastAPI) -> None:
+    """Add exception handlers to the FastAPI application.
+
+    Args:
+        app: the FastAPI application.
+
+    Returns:
+        None
+    """
+    app.add_exception_handler(StarletteHTTPException, starlette_exception_handler)
+
+    app.add_exception_handler(RequestError, eodag_errors_handler)
```

Comment:
Argument 2 to "add_exception_handler" of "Starlette" has incompatible type "Callable[[Request, EodagError], Coroutine[Any, Any, ORJSONResponse]]"; expected "Callable[[Request, Exception], Response | Awaitable[Response]] | Callable[[WebSocket, Exception], Awaitable[None]]"mypy[arg-type](https://mypy.readthedocs.io/en/stable/_refs.html#code-arg-type)